impl/sql/appender/objhasher.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appender
+
+import (
+	"crypto"
+
+	"github.com/benlaurie/objecthash/go/objecthash"
+)
+
+// Domain separation prefixes
+const (
+	RFC6962LeafHashPrefix = 0
```

Comment:
Good question.  These prefixes provide "domain separation" in the Merkle Tree, preventing leaf hashes from being glued to interior nodes etc. 

ObjectHash uses prefixes as well to distinguish types. None of these prefixes use `1` as a value, so I believe we retain the domain separation property. 

I'm not sure it makes sense to try to add an additional domain separator to the output of a hash.  We'd end up with a double hash like `H(1 || ObjectHash(obj))`